src/develop/imageop.c

```diff
@@ -1696,6 +1807,9 @@ void dt_iop_commit_params(dt_iop_module_t *module, dt_iop_params_t *params,
     // register if module allows tiling, commit_params can overwrite this.
     if(module->flags() & IOP_FLAGS_ALLOW_TILING) piece->process_tiling_ready = 1;
 
+    if(module->so->get_introspection())
+      _iop_validate_params(module->so->get_introspection()->field, params, TRUE);
```

Comment:
This is run for every commit_params() ? it probably slow down a bit dt then as there is many commit_params calls especially when dragging a curve for example. Can't we do that only when leaving darkrroom or changing the image ?

Reply:
> This is run for every commit_params() ? it probably slow down a bit dt then as there is many commit_params calls especially when dragging a curve for example. Can't we do that only when leaving darkrroom or changing the image ?

The point would be to determine if crashes are caused by faulty input or by something later in the calculations. For that, everything passed to commit_params would need to be checked. This step is very lightweight compared to full image processing, but could be more impactful if most calculations of the pipe are cached/skipped.

Would a debugging flag be more useful? -d check_params? So then the process would be that if someone reports an unexplained crash, they are then asked to run using this flag and see if there is any output. Alternatively, this could be switched on automatically if the last dt session crashed (left behind a "lock" file that gets deleted on normal exit). Any errors should then be written to a log file that we can request an error-reporter to upload. The risk here is that some people might always end a session by killing it; they would then always run with debug settings.

Reply:
For hardening purposes it could be sufficient to just check params loaded from xmp/db.